cli/bin/semgrep

```diff
@@ -66,38 +72,42 @@ def find_semgrep_core_path():
           file=sys.stderr)
     # fatal error, see src/osemgrep/core/Exit_code.ml
     sys.exit(2)
-       
 
-# when python imports a module, i.e. `import semgrep.FOO`, it runs the module's
+# We could have moved the code below in a separate 'osemgrep' file, like
+# for 'pysemgrep', but we don't want users to be exposed to another command,
+# so it is better to hide it.
+# We expose 'pysemgrep' because osemgrep itself might need to fallback to
+# pysemgrep and it's better to avoid the possibility of an infinite loop
+# by simply using a different program name. Morever, in case of big problems,
+# we can always tell users to run pysemgrep instead of semgrep and be sure
+# they'll get the old behavior.
+def call_osemgrep():
+    path = find_semgrep_core_path()
+    # If you call semgrep-core as osemgrep, then we get osemgrep behavior,
+    # see src/main/Main.ml
+    sys.argv[0] = "osemgrep"
+    os.execvp(str(path), sys.argv)
```

Comment:
I think that a lot of the errors in the tests are because we're not passing environment variables here, which is important to not break semgrep in CI.

Reply:
Actually checking the docs, it seems execvp will inherit environment variables by default so I think this is fine

Reply:
yes, children inherits the environment of the parent (UNIX got many things right!)